dd-sdk-android-ndk/src/main/cpp/datadog-native-lib.cpp

```diff
@@ -34,11 +49,23 @@ std::string get_serialized_log(int signal,
                                const char *backtrace) {
     std::string serializedLog;
     serializedLog.append("{ ");
-    serializedLog.append(R"("message": ")").append(GENERIC_MESSAGE).append("\",");
-    serializedLog.append(R"("service": ")").append(SERVICE_NAME).append("\",");
-    serializedLog.append(R"("logger.name": ")").append(LOGGER_NAME).append("\",");
-    serializedLog.append(R"("env": ")").append(ENVIRONMENT).append("\",");
-    serializedLog.append(R"("status": ")").append(EMERGENCY_STATUS).append("\",");
+    if (rumcontext.application_id[0] != '\0') {
```

Comment:
If the context is not initialized, nothing guarantees you that the first byte will be set to `\0`. I haven't read the rest of the code, but do you have some logic somewhere that sets the `char[]` fields to 0?

Reply:
That's actually a question I wanted to ask you. When I declared before :
```
static struct rumcontext {
    char application_id[200];
    char session_id[200];
    char view_id[200];
} rumcontext;

```
This is not equivalent with:

```
static struct rumcontext {
    char application_id[200];
    char session_id[200];
    char view_id[200];
} rumcontext{};

```
??

And also this doesn't mean allocating the memory and adding `\0` for all the bytes ?